add clarifying language about using items instead of item (#172)
radiantearth/stac-api-spec: overview.md

| `/api`                                              | [OAFeat](ogcapi-features)  | service-desc      | OpenAPI 3.0 JSON                                  | The OpenAPI definition of the endpoints in this service                                                                             |
| **`/collections/{collectionId}`**                   | [OAFeat](ogcapi-features)  | collection        | Collection                                        | Returns single Collection JSON                                                                                                      |
| **`/collections/{collectionId}/items`**             | [OAFeat](ogcapi-features)  | items             | ItemCollection                                    | GeoJSON FeatureCollection-conformant entity of Item objects in collection                                                                  |

> | **`/collections/{collectionId}/items/{featureId}`** | [OAFeat](ogcapi-features)  | item              | Returns single Item (GeoJSON Feature). This relation is most commonly used in static STAC catalogs. |

Reviewer: ```suggestion
| **`/collections/{collectionId}/items/{featureId}`** | [OAFeat](ogcapi-features)  | item              | Returns single Item (GeoJSON Feature). This relation is usually not used in OAFeat implementations. |
```

item is used in browsable APIs, too.

> | **`/collections/{collectionId}/items/{featureId}`** | [OAFeat](ogcapi-features)  | item              | Returns single Item (GeoJSON Feature). This relation is defined by STAC, but its use is discouraged in STAC API, as Items should be referenced through their containing Collection rather than directly.            |                                                                                                                                     |

Reviewer 2: What does this mean? That items should be referenced through their containing Collection? What's the reason to discourage this?

Reviewer 3: I'm not sure it should be discouraged per se, you could have a browsable API that has both child catalogs (and ultimately items).  Or like @fredliporace has implemented I think, where the whole catalog is a normal static catalog with only the /search endpoint being dynamic.

Reviewer 4: @matthewhanson yes, the [API](https://stac.amskepler.com/v100) includes a child link to the static catalog. Did this to list a single entry in stacindex, that references both the search endpoint and the static catalog.

Reviewer: Yes, I agree. This should not be discouraged.
I think I'd say static catalogs should link to the API, too. And APIs should link to static catalogs, so that no matter where you start, you can find everything.

Author: I rewrote this in the README to be more clear -- I feel like I'm having a hard time expressing this well. please re-review.
